src/providers/ldap/sdap.c

```diff
@@ -1614,10 +1614,14 @@ sdap_get_primary_fqdn(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    primary_fqdn = sss_create_internal_fqname(tmp_ctx, shortname, dom->name);
-    if (primary_fqdn == NULL) {
-        ret = ENOMEM;
-        goto done;
+    if (strchr(shortname, '@') == NULL) {
```

Comment:
I wonder if this check is safe for the future. We already have a request to support  group names with @ in them, so I think this check could cause some false positives in the future. I wonder if it was safer to always parse the name and also pass in the "expected" domain and compare if the parsed and expected domain match. I think if the function only performed a check and didn't return anything except a true/false boolean, we could even avoid memory allocation, so the performance impact would be minimal.